nvi

```diff
@@ -10,7 +10,12 @@ N=$(tput sgr0) # Normal
 U=$(tput smul) # Underline
 
 print_usage() {
-  cat << EOF
+  less \
+    --raw-control-chars \
+    --no-init \
+    --quit-if-one-screen \
+    --QUIT-AT-EOF \
```

Comment:
The main reason the current `--help` text isn't a regular man page is because my search for how to create one was exceptionally discouraging. The guides all told of strange, old, convoluted macro languages and other crimes against readability. I like the idea of having a fully-fledged man page, and writing a shorter (ideally one-page in a 80×24 terminal emulator) text for `--help`.

The current implementation went with a faux man page because, as you say, the man page format is what you typically reach for when you need documentation for a CLI tool. Having only a regular `--help` with the typical brevity and casual formatting would be a significant loss of documentation quality.

I would love to have a proper man page, but I imagine it will force the user to download more than one file. Currently you can just download `https://github.com/DanskSupermarked/nvi/releases/download/1.0.0/nvi` in your CI environment and be all set. I really like this simple contract as it resonates so well with the conceptual purpose of `nvi`: to make installing Node.js in CI environments as painless as possible.

I'll list a few ideas for how documentation of `nvi` could work here. They're ordered by descending appeal to me, and they are probably not all possible:

1.  We somehow embed the man page within the single `nvi` file and provide both `man nvi` (with the current help text) and `nvi --help` (with a short summary). I don't think this is technically possible.
2.  We make `nvi help` render a fully-fledged man page with the current help text, and rewrite the text for `nvi --help` as a summary. We do this without requiring the user to download additional files beyond `nvi`. I imagine the implementation needed will be terribly ugly.
3.  We make `nvi help` render a _man page-like_ help text using `less` or friends, and rewrite the text for `nvi --help` as a summary. We do this without requiring the user to download additional files beyond `nvi`.
4.  We create a regular man page and ship `nvi` as an archive instead of a single file. We'll then have `man nvi` for the full text and `nvi --help` for the summary.

How far down the list do you think we need to drop before it becomes a possible and feasible task to implement?